eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg

```diff
@@ -59,6 +59,29 @@ public interface Immutable<name>Set extends Immutable<name>Collection, <name>Set
         }
     }
 
+    /**
+     * Returns the set of all objects that are members of both {@code this} and {@code set}. The intersection of
+     * [1, 2, 3] and [2, 3, 4] is the set [2, 3].
+     *
+     * @since 11.0.
+     */
+    @Override
+    default Immutable<name>Set intersect(<name>Set set)
+    {
+        if (this.size() \< set.size())
+        {
+            Mutable<name>Set mutableSet = this.toSet();
+            mutableSet.retainAll(set);
```

Comment:
use `intersect` instead of `retainAll` (and below)

Reply:
Won't this create two copies of the set then?

Actually, I think we can use the same approach as above... use `select` directly which will return an `ImmutableSet`.

Reply:
Used `select` here as well.